packages/amp/src/processors/img.tsx

```diff
@@ -0,0 +1,51 @@
+import { Processor, Element } from "@frontity/html2react/types";
+import { Packages } from "../../types";
+
+/**
+ * Props for the image element as processed by {@link img}.
+ */
+type ImgElement = Element & {
+  /**
+   * Props.
+   */
+  props: {
+    /**
+     * CSS classes.
+     */
+    class: string;
+
+    /**
+     * Loading prop that normally is set on the img component by WordPress.
+     */
+    loading: boolean;
+
+    /**
+     * Layout - refers to AMP layout parameter.
+     */
+    layout: string;
+  };
+};
+
+export const img: Processor<ImgElement, Packages> = {
+  name: "amp-img",
+  test: ({ node }) => node.type === "element" && node.component === "img",
+  processor: ({ node }) => {
+    // Because amp-* component are custom components, we have to pass the
+    // class as `class` and not as `className`:
+    // https://reactjs.org/docs/web-components.html#using-web-components-in-react
+    node.props.class = node.props?.className;
+
+    // We can now safely remove the className prop.
+    delete node.props.className;
+
+    // amp-img does not support nor need that prop.
+    delete node.props.loading;
```

Comment:
Is this true only for `amp-img` or also for `amp-iframe`? If so, should we add it to the `iframe` processor as well?

Reply:
Yes, the amp-iframe also does not allow the `loading` attribute, but the difference here is that WordPress will automatically add the `loading="lazy"` attribute on the images so we have to strip that. 

However the when creating an iframe the user would have to opt-in to using the `loading="lazy"` in their content in which case the AMP validator which runs in development in all AMP sites would fail informing them that this property is not allowed.

So, I think we can't go wrong either way, but maybe it would be slightly better to NOT remove the `loading` and just let the user fix their content. What do you think?

Reply:
> the difference here is that WordPress will automatically add the loading="lazy" attribute on the images

Yeah, but that's the current version. Maybe they also add native lazy loading to iframes in the next version. Or in the iframe Gutenberg block. Or maybe users use a plugin for that, like https://wordpress.org/plugins/lazy-loading-responsive-images/...

> maybe it would be slightly better to NOT remove the loading and just let the user fix their content. What do you think?

The problem is that the `post.content` is shared between the React version and the AMP version. If they remove it for AMP, they would be removing it from React as well.

Reply:
> The problem is that the post.content is shared between the React version and the AMP version. If they remove it for AMP, they would be removing it from React as well.


That's a fair point! Will remove the `loading` in the processor in that case.

Reply:
Awesome. Thanks Michal!